src/com/esotericsoftware/yamlbeans/Beans.java

```diff
@@ -38,253 +38,370 @@
+				if (setMethod != null || constructorProperty) {
 					return new MethodProperty(name, setMethod, getMethod);
+				}
 			}
 		}
 
+		FieldProperty fieldProperty = null;
 		for (Field field : getAllFields(type)) {
-			if (!field.getName().equals(name)) continue;
-			int modifiers = field.getModifiers();
-			if (Modifier.isStatic(modifiers) || Modifier.isTransient(modifiers)) continue;
-			if (!Modifier.isPublic(modifiers) && !privateFields) continue;
-			try {
+			if (!field.getName().equals(name)) {
+				continue;
+			}
+
+			if (isProperField(field, privateFields)) {
 				field.setAccessible(true);
-			} catch (Exception ignored) {
+				fieldProperty = new FieldProperty(field);
+				break;
+			}
+		}
+
+		return fieldProperty;
+	}
+
+	static private String toJavaIdentifier(String name) {
+		StringBuilder builder = new StringBuilder();
+
+		for (int i = 0; i < name.length(); i++) {
+			char c = name.charAt(i);
+
+			if (Character.isJavaIdentifierPart(c)) {
+				builder.append(c);
+			}
+		}
+
+		return builder.toString();
+	}
+
+	static private Constructor getNoArgConstructor(Class type) {
+		Constructor noArgConstructor = null;
+
+		for (Constructor typeConstructor : type.getConstructors()) {
+			if (isNoArgConstructor(typeConstructor)) {
+				noArgConstructor = typeConstructor;
+				break;
+			}
+		}
+
+		return noArgConstructor;
+	}
+
+	static private boolean isNoArgConstructor(Constructor constructor) {
+		return constructor.getParameterTypes().length == 0;
+	}
+
+	static private Constructor getPrivateConstructor(Class type) {
+		Constructor privateConstructor = null;
+
+		try {
+			privateConstructor = type.getDeclaredConstructor();
+			privateConstructor.setAccessible(true);
+		} catch (SecurityException ignored) {
+		} catch (NoSuchMethodException ignored) {
+		}
+
+		return privateConstructor;
+	}
+
+	static private Constructor tryCommonImplementationConstructor(Class type) throws InvocationTargetException {
+		Constructor commonImplementationConstructor = null;
+		Class[] noArgs = new Class[0];
+
+		try {
+			if (List.class.isAssignableFrom(type)) {
+				commonImplementationConstructor = ArrayList.class.getConstructor(noArgs);
+			} else if (Set.class.isAssignableFrom(type)) {
+				commonImplementationConstructor = HashSet.class.getConstructor(noArgs);
+			} else if (Map.class.isAssignableFrom(type)) {
+				commonImplementationConstructor = HashMap.class.getConstructor(noArgs);
 			}
-			return new FieldProperty(field);
+		} catch (Exception e) {
+			throw new InvocationTargetException(e, "Error getting constructor for class: " + type.getName());
 		}
-		return null;
+
+		return commonImplementationConstructor;
 	}
 
-	static private ArrayList<Field> getAllFields (Class type) {
-		ArrayList<Class> classes = new ArrayList();
+	static private ArrayList<Field> getAllFields(Class type) {
+		ArrayList<Class> classes = new ArrayList<Class>();
+		ArrayList<Field> allFields = new ArrayList<Field>();

```

Comment:
Moving all declarations first hurts readability will not be merged. Are you using automated tools to make these changes?